src/Components/SampleShowcase/SampleShowcase.tsx

```diff
@@ -141,36 +160,91 @@ export class SampleShowcase extends React.Component<{}, ShowcaseState> {
     this.setupNewSample(groupName, sampleName);
   }
 
-  private onIModelChange = (iModelName: string) => {
+  private _onIModelChange = (iModelName: string) => {
     this.setState({ iModelName }, () => this._onActiveSampleChange(this.state.activeSampleGroup, this.state.activeSampleName));
   }
 
-  private onEditorButtonClick() {
+  private _onEditorButtonClick = () => {
     this.setState((prevState) => ({ showEditor: !prevState.showEditor }));
   }
 
+  private _onSampleTranspiled = async (blob: string) => {
+    const activeSample = this.getSampleByName(this.state.activeSampleGroup, this.state.activeSampleName)!;
+    const sampleUi = await import( /* webpackIgnore: true */ blob);
+
+    if (!this._prevSampleSetup) {
+      this._prevSampleSetup = activeSample.setup;
+    }
+    if (!this._prevSampleTeardown) {
+      this._prevSampleTeardown = activeSample.teardown;
+    }
+
+    /* HACK: Because some samples have a setup method stored in the sample itself, which is editable, we need to make suree call it.
```

Comment:
We could refactor the samples so that the setup / teardown methods always live in the SampleSpec.tsx.  For some of them, those methods might need to call a method in the SampleApp class (because the setup and teardown logic is important for the user to see).  Would that help?

Reply:
I don’t think that would help much in this case, unless we also load the “App” portion of the sample as a “module” by referencing it in the modules.json. This would make the “App” portion of the sample uneditable, which I think starts taking away from the experience. What might work instead is if we created an interface and implemented it for each of the “App” portions of the sample. Then when it’s time to import, we can cast the default import as that interface. That would solve the need to search for the right method.

Probably something I should’ve done instead of this.

Reply:
>>if we created an interface and implemented it for each of the “App” portions of the sample

That is quite reasonable.

Reply:
I've updated to use a "SampleApp" abstract class that each sample implements. I'm able to cast the import to this type and it works nicely to allow the showcase to call the correct method.